src/Components/3DV/SceneView.stories.tsx

```diff
@@ -16,6 +35,231 @@ export default {
+        (gizmoTransformItem: TransformInfo) => {
+            dispatch({
+                type: SET_GIZMO_TRANSFORM_ITEM,
+                payload: gizmoTransformItem
+            });
+            console.log(
+                'new position: ',
+                gizmoTransformItem.position,
+                '\nnew rotation: ',
+                gizmoTransformItem.rotation
+            );
+        },
+        []
+    );
+
+    const setGizmoElementItem = useCallback(
+        (gizmoElementItem: TransformedElementItem) => {
+            dispatch({
+                type: SET_ELEMENT_TO_GIZMO,
+                payload: gizmoElementItem
+            });
+        },
+        []
+    );
+
+    const onGizmoChange = useCallback((event, checked?: boolean) => {
+        if (!checked) {
+            setGizmoElementItem({ parentMeshId: null, meshIds: null });
+        } else {
+            setGizmoElementItem(defaultGizmoElementItems);
+        }
+        setIsGizmo(!!checked);
+    }, []);
+
+    return (
+        <div style={{ position: 'relative' }}>
+            <div>
+                <Checkbox
+                    label="Show Gizmo and Transform"
+                    checked={isGizmo}
+                    onChange={onGizmoChange}
+                />
+            </div>
+            <div>
+                <SceneView
+                    modelUrl="https://cardboardresources.blob.core.windows.net/cardboard-mock-files/OutdoorTanks.gltf"
+                    gizmoElementItem={
+                        state.gizmoElementItem ?? defaultGizmoElementItems
+                    }
+                    setGizmoTransformItem={setGizmoTransformItem}
+                    gizmoTransformItem={state.gizmoTransformItem}
+                />
+            </div>
+            <div style={{ position: 'absolute', bottom: 0 }}>
+                {isGizmo && (
+                    <FakePanel
+                        gizmoTransformItem={state.gizmoTransformItem}
+                        setGizmoTransformItem={setGizmoTransformItem}
+                    ></FakePanel>
+                )}
+            </div>
+        </div>
+    );
+};
+
+interface IFakePanelProps {
+    gizmoTransformItem: TransformInfo;
+    setGizmoTransformItem?: (gizmoTransformItem: TransformInfo) => void;
+}
+
+const FakePanel = (props: IFakePanelProps) => {
+    const { gizmoTransformItem, setGizmoTransformItem } = props;
+    const gizmoTransformItemDraftRef = useRef<TransformInfo>(null);
+
+    useEffect(() => {
+        if (gizmoTransformItem) {
+            gizmoTransformItemDraftRef.current = deepCopy(gizmoTransformItem);
+        }
+    }, [gizmoTransformItem]);
+
+    const onTransformChange = useCallback(
+        (event) => {
+            if (gizmoTransformItem) {
+                const { name, value } = event.target;
+                gizmoTransformItemDraftRef.current = deepCopy(
+                    gizmoTransformItem
+                );
+                const valueAsNumber = value;
```

Comment:
what is the reason for this new variable declaration? can't we just use the `value` for below assignments?

Reply:
ah that was leftover from something i was doing before -- will fix